File: drawing-java/brushes.js
class Brush {
    constructor(settings) {
        this.points = [];
        this.settings = settings;
    }

    addPoint(point) {
        this.points.push(point);
    }

    draw(context) {
        if (this.points.length > 1) {
            context.beginPath();
            context.strokeStyle = this.settings.color;
            context.lineWidth = this.settings.lineThickness;
            context.moveTo(this.points[0].x, this.points[0].y);
            
            for (let i = 1; i < this.points.length; i++) {
                context.lineTo(this.points[i].x, this.points[i].y);
            }
            
            context.stroke();
        }
    }
}

class Highlighter extends Brush {
    draw(context) {
        if (this.points.length > 1) {
            context.beginPath();
            context.globalAlpha = this.settings.transparency;
            context.strokeStyle = this.settings.color;
            context.lineWidth = this.settings.lineThickness;
            context.moveTo(this.points[0].x, this.points[0].y);
            
            for (let i = 1; i < this.points.length; i++) {
                context.lineTo(this.points[i].x, this.points[i].y);
            }
            
            context.stroke();
            context.globalAlpha = 1.0;
        }
    }
}

class Brushes {
    constructor() {
        this.brushes = new Map();
        this.currentBrush = null;
        this.currentHighlighter = null;
    }

    createBrush(settings) {
        const brush = new Brush(settings);
        this.brushes.set('brush', brush);
        this.currentBrush = brush;
    }

    createHighlighter(settings) {
        const highlighter = new Highlighter(settings);
        this.brushes.set('highlighter', highlighter);
        this.currentHighlighter = highlighter;
    }

    addPoint(point) {
        if (this.currentBrush) {
            this.currentBrush.addPoint(point);
        }
        if (this.currentHighlighter) {
            this.currentHighlighter.addPoint(point);
        }
    }

    draw(context) {
        if (this.currentBrush) {
            this.currentBrush.draw(context);
        }
        if (this.currentHighlighter) {
            this.currentHighlighter.draw(context);
        }
    }

    clearPoints() {
        this.currentBrush = null;
        this.currentHighlighter = null;
        this.brushes.clear();
    }
}

export { Brush, Highlighter, Brushes };

File: drawing-logic/brushes.py
import pygame
from .settings import drawing_settings

class Brush:
    def __init__(self):
        self.points = []

    def add_point(self, point):
        self.points.append(point)

    def draw(self, surface):
        if len(self.points) > 1:
            pygame.draw.lines(
                surface, 
                drawing_settings.color, 
                False, 
                self.points, 
                drawing_settings.line_thickness
            )

class Highlighter(Brush):
    def __init__(self):
        super().__init__()

    def draw(self, surface):
        if len(self.points) > 1:
            line_surface = pygame.Surface(surface.get_size(), pygame.SRCALPHA)
            color = drawing_settings.color + (int(drawing_settings.transparency * 255),)
            pygame.draw.lines(
                line_surface, 
                color, 
                False, 
                self.points, 
                drawing_settings.line_thickness
            )
            surface.blit(line_surface, (0, 0))

class Brushes:
    def __init__(self):
        self.brushes = {}
        self.current_brush = None
        self.current_highlighter = None

    def create_brush(self):
        brush = Brush()
        self.brushes['brush'] = brush
        self.current_brush = brush

    def create_highlighter(self):
        highlighter = Highlighter()
        self.brushes['highlighter'] = highlighter
        self.current_highlighter = highlighter

    def add_point(self, point):
        if self.current_brush:
            self.current_brush.add_point(point)
        if self.current_highlighter:
            self.current_highlighter.add_point(point)

    def draw(self, surface):
        if self.current_brush:
            self.current_brush.draw(surface)
        if self.current_highlighter:
            self.current_highlighter.draw(surface)

    def clear_points(self):
        if self.current_brush:
            self.current_brush.points = []
        if self.current_highlighter:
            self.current_highlighter.points = []